tests/testthat/test-tunable.R

```diff
@@ -0,0 +1,145 @@
+  expect_true(all(c5_info$component == "boost_tree"))
+  expect_true(all(c5_info$component_id[1:3] == "main"))
+  expect_true(all(c5_info$component_id[-(1:3)] == "engine"))
+  nms <- c("trees", "min_n", "sample_size", "rules", "CF", "noGlobalPruning",
+           "winnow", "fuzzyThreshold", "bands")
+  expect_equal(c5_info$name, nms)
+  expect_true(all(purrr::map_lgl(c5_info$call_info[1:3], ~ .x$pkg == "dials")))
+  expect_equal(
+    purrr::map_chr(c5_info$call_info[1:3], ~ .x$fun),
+    c("trees", "min_n", "sample_prop")
+  )
+  expect_true(sum(purrr::map_lgl(c5_info$call_info, is.null)) == 1)
+})
+
+
+test_that('bad model inputs', {
+  lm_model <- linear_reg() %>% set_engine("lm")
+
+  bad_class <- lm_model
+  class(bad_class) <- c("potato", "model_spec")
+  expect_error(
+    tunable(bad_class),
+    "model database doesn't know about the arguments for model"
+  )
+})
+
+test_that("workflow with tunable recipe", {
+  spline_rec <- recipe(ridership ~ ., data = head(Chicago)) %>%
+    step_date(date) %>%
+    step_holiday(date) %>%
+    step_rm(date, ends_with("away")) %>%
+    step_impute_knn(all_predictors(), neighbors = tune("imputation")) %>%
+    step_other(all_nominal(), threshold = tune()) %>%
+    step_dummy(all_nominal()) %>%
+    step_normalize(all_predictors()) %>%
+    step_bs(all_predictors(), deg_free = tune(), degree = tune())
+  lm_model <- linear_reg() %>%
+    set_engine("lm")
+  wf_tunable_recipe <- workflow(spline_rec, lm_model)
+
+  wf_info <- tunable(wf_tunable_recipe)
+  check_tunable_tibble(wf_info)
+  expect_true(all(wf_info$source == "recipe"))
+})
+
+test_that("workflow with tunable model", {
+  rm_rec <- recipe(ridership ~ ., data = head(Chicago)) %>%
+    step_rm(date, ends_with("away"))
+  bst_model <-
+    boost_tree(mode = "classification", trees = tune("funky name \n")) %>%
+    set_engine("C5.0", rules = tune(), noGlobalPruning = TRUE)
+  wf_tunable_model <- workflow(rm_rec, bst_model)
+
+  wf_info <- tunable(wf_tunable_model)
+  check_tunable_tibble(wf_info)
+  expect_equal(nrow(wf_info), 9)
+  expect_true(all(wf_info$source == "model_spec"))
+})
+
+test_that("workflow with tunable recipe and model", {
+  spline_rec <- recipe(ridership ~ ., data = head(Chicago)) %>%
+    step_date(date) %>%
+    step_holiday(date) %>%
+    step_rm(date, ends_with("away")) %>%
+    step_impute_knn(all_predictors(), neighbors = tune("imputation")) %>%
+    step_other(all_nominal(), threshold = tune()) %>%
+    step_dummy(all_nominal()) %>%
+    step_normalize(all_predictors()) %>%
+    step_bs(all_predictors(), deg_free = tune(), degree = tune())
+  bst_model <-
+    boost_tree(mode = "classification", trees = tune("funky name \n")) %>%
+    set_engine("C5.0", rules = tune(), noGlobalPruning = TRUE)
+  wf_tunable <- workflow(spline_rec, bst_model)
+
+  wf_info <- tunable(wf_tunable)
+  check_tunable_tibble(wf_info)
+  expect_equal(
+    wf_info$source,
+    c(rep("model_spec", 9), rep("recipe", 4))
+  )
+})
+
```

Comment:
Maybe one test about a workflow without any tunable parameters? I assume that would work fine, but one test for it couldn't hurt